sktime/forecasting/naive.py

```diff
@@ -139,5 +154,53 @@ def _predict_last_window(self, fh, X=None, return_pred_int=False,
             fh_idx = fh.index_like(self.cutoff)
             return last_window[fh_idx]
 
-        elif self.strategy == "mean":
+        elif self.strategy == "mean" and self.sp == 1:
             return np.repeat(np.nanmean(last_window), len(fh))
+
+        elif self.strategy == "mean" and self.sp != 1:
+            last_window = last_window.astype(float)
+            last_window = np.pad(last_window,
+                                 (0, self.sp_ - len(last_window) % self.sp_),
+                                 'constant',
+                                 constant_values=(-1))
+
+            last_window = last_window.reshape(np.int(np.
+                                                     ceil(len(last_window) /
+                                                          self.sp_)),
+                                              self.sp_)
+
+            indices = np.where(last_window == -1)[-1]
+            last_window[-1][indices] =\
+                (last_window.sum(axis=0)[indices] +
+                 1) / (last_window.shape[-1] - 1)
+
+            last_window = last_window.mean(axis=0)
+
+            # we need to replicate the last window if max(fh) is
+            # larger than sp,
+            # so that we still make forecasts by repeating the
+            # last value for that season,
+            # assume fh is sorted, i.e. max(fh) == fh[-1]
+            # only slicing all the last seasons into last_window
+            if fh[-1] > self.sp_:
+                reps = np.int(np.ceil(fh[-1] / self.sp_))
+                last_window =\
+                    np.tile(last_window,
+                            reps=reps)
+
+            # get zero-based index by subtracting the minimum
+            fh_idx = fh.index_like(self.cutoff)
+            return last_window[fh_idx]
+
+        elif self.strategy == "drift":
+            if any(last_window) is not np.nan:
```

Comment:
what happens if this condition fails? perhaps we could just use `np.nanmean` to handle missing values? Or raise an error?